gensim/models/word2vec.py

```diff
@@ -654,27 +654,27 @@ def finalize_vocab(self):
             # create null pseudo-word for padding when using concatenative L1 (run-of-words)
             # this word is only ever input – never predicted – so count, huffman-point, etc doesn't matter
             word, v = '\0', Vocab(count=1, sample_int=0)
-            v.index = len(self.vocab)
-            self.index2word.append(word)
-            self.vocab[word] = v
+            v.index = len(self.kv.vocab)
+            self.kv.index2word.append(word)
+            self.kv.vocab[word] = v
         # set initial input/projection and hidden weights
         self.reset_weights()
 
     def sort_vocab(self):
         """Sort the vocabulary so the most frequent words have the lowest indexes."""
-        if hasattr(self, 'syn0'):
+        if self.kv.syn0:
```

Comment:
I'm not completely sure this'll work since right now, `self.syn0` is initialized in `reset_weights`, which is called after `sort_vocab`. With `kv` being assigned in `__init__`, wouldn't this always raise an exception?
Possibly have a `reset_weights` equivalent in `KeyedVectors`?